flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -109,10 +112,20 @@ class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
             mainIsolateStart: mainIsolateStartDateTime);
 
         setAppStartInfo(appStartInfo);
-      });
-    }
+      } else {
+        // We are not adding the app start end time, since it might be set later by the user
+        // through SentryFlutter.setAppStartEnd and is going to be queried in the event processor
+        final appStartInfo = AppStartInfo(
+            nativeAppStart.isColdStart ? AppStartType.cold : AppStartType.warm,
+            start: appStartDateTime,
+            pluginRegistration: pluginRegistrationDateTime,
+            mainIsolateStart: mainIsolateStartDateTime);
+
+        setAppStartInfo(appStartInfo);
+      }
+    });
```

Comment:
I'd put this whole section after the `if{}` block, removing the `else`.
Just initialize `appStartEndDateTime` to null and set a value in the `if` block.
So we avoid repetition and an `else` branch